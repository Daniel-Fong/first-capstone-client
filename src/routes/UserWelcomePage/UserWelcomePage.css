.welcome-page-main {
    display: inline-flex;
    width: 100%;
    justify-content: space-around;
}

.user-welcome-div h1 {
    font-family: cursive;
}

.games-div {
    width: 40%;
}

.players-div {
    width: 40%;
}

body {
    background-color:rgb(47, 52, 56);
    color: white;
}

@media screen and (max-width: 600px) {
    .welcome-page-main input {
        width: 10em;
    }
}