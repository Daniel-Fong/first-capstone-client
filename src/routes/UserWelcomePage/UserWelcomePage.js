import React from 'react'
import GamesList from '../../components/GamesList/GamesList'
import config from '../../config'
import TokenService from '../../services/token-service'
import PlayersList from '../../components/PlayersList/PlayersList'
import { Link } from 'react-router-dom'
import CreateGameForm from '../../components/CreateGameForm/CreateGameForm'

class UserWelcomePage extends React.Component {
    state = {
        games: [],
        players: [],
        error: null,
    }

    componentDidMount() {
       this.fetchGamesByUserId()
        .then((games) => {
            this.setState({games})
        })
        .catch(error => {
            console.error({error})
        })
        this.fetchPlayersByUserId()
        .then((players) => {
            this.setState({players})
        })
        .catch(error => {
            console.error({error})
        })
    }

    handleAddGame = (e) => {
        const name = e.target.name.value
        const notes = e.target.notes.value
        let newGame = JSON.stringify({
            name: name,
            notes: notes
        })
        return(
          fetch(`http://localhost:8080/api/games`, {
            method: 'POST',
            headers: {
              'authorization': `bearer ${TokenService.getAuthToken()}`,
              'Content-type': 'application/json'
            },
            body: newGame
          })
          .then(res => {
            if (!res.ok)
              return res.json().then(e => Promise.reject(e))
            return res.json()
          })
          .then(game => {
            this.setState({games: [...this.state.games, game]})
          })
          .catch(error => {
            console.error({ error })
          })
          )}

    handleAddPlayer = (e) => {
      const name = e.target.name.value
        const notes = e.target.notes.value
        let newPlayer = JSON.stringify({
            name: name,
            notes: notes
        })
        return(
          fetch(`http://localhost:8080/api/players`, {
            method: 'POST',
            headers: {
              'authorization': `bearer ${TokenService.getAuthToken()}`,
              'Content-type': 'application/json'
            },
            body: newPlayer
          })
          .then(res => {
            if (!res.ok)
              return res.json().then(e => Promise.reject(e))
            return res.json()
          })
          .then(player => {
            this.setState({players: [...this.state.players, player]})
          })
          .catch(error => {
            console.error({ error })
          })
        )}

    handleDeleteGameById = (gameId) => {
        return(
          fetch(`http://localhost:8080/api/games/${gameId}`, {
            method: 'DELETE',
            headers: {
              'authorization': `bearer ${TokenService.getAuthToken()}`,
              'Content-type': 'application/json'
            }
          })
            .then(res => {
              this.setState({games: this.state.games.filter(game=> game.id !== gameId)})
            })         
        )}

    handleDeletePlayerById = (playerId) => {
        return(
          fetch(`http://localhost:8080/api/players/${playerId}`, {
            method: 'DELETE',
            headers: {
              'authorization': `bearer ${TokenService.getAuthToken()}`,
              'Content-Type': 'application/json'
            }
          })
              .then(res => {
                this.setState({players: this.state.notes.filter(player=> player.id !== playerId)})
              })         
        )}

    fetchPlayersByUserId() {
        return(
            fetch(`${config.API_ENDPOINT}/players`, {
                headers: {
                    'authorization': `bearer ${TokenService.getAuthToken()}`,
                },
            })
              .then(res => {
                if (res.ok) {
                  return res.json();
                }
                return Promise.reject('Error fetching players from server');
              })
              .catch(err => {
                this.setState({error: err})
              })
            )
          }
    
    fetchGamesByUserId() {
        return(
          fetch(`${config.API_ENDPOINT}/games`, {
              headers: {
                  'authorization': `bearer ${TokenService.getAuthToken()}`,
              },
          })
            .then(res => {
              if (res.ok) {
                return res.json();
              }
              return Promise.reject('Error fetching games from server');
            })
            .catch(err => {
              this.setState({error: err})
            })
          )
        }

    render() {
        return (
            <div className='user-welcome-div'>
                <header>
                    <ul className='navbar'>
                        <Link to='/users'>
                        <li>Home</li>
                        </Link>
                        <li>My Games</li>
                        <li><button>Create New Game</button></li>
                        <CreateGameForm handleAddGame={this.handleAddGame} />
                    </ul>
                    <h2>Welcome User</h2>
                </header>
                <main>
                    <button>Start a New Game</button>
                    <h3>My Saved Games</h3>
                    <label>Sort By</label>
                    <select>
                        <option>Date Old to New</option>
                        <option>Date New to Old</option>
                    </select>
                    <label>Search</label>
                    <input />
                    <div>
                        <GamesList games={this.state.games} userId={this.state.userid} handleDeleteGame={this.handleDeleteGameById} />
                        <PlayersList render={false} players={this.state.players} userId={this.state.userid} handleDeletePlayer={this.handleDeletePlayerById} />
                    </div>
                </main>
            </div>
        )
    }
}

export default UserWelcomePage;