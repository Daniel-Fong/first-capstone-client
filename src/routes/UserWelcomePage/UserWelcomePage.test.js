import React from 'react'
import ReactDOM from 'react-dom';
import UserWelcomePage from './UserWelcomePage'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'

describe('<UserWelcomePage />', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
        <Router>
            <UserWelcomePage match={{params: {gameid: 1}}}/>
        </Router>    
            , div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it ('renders the ui as expected', () => {
        const mockUserWelcomePage= renderer
            .create(
            <Router>
                <UserWelcomePage match={{params: {gameid: 1}}}/>
            </Router>)
            .toJSON();
        expect(mockUserWelcomePage).toMatchSnapshot();
    })
})