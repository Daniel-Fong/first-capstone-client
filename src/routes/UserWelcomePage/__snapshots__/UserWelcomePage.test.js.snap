// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserWelcomePage /> renders the ui as expected 1`] = `
<div
  className="user-welcome-div"
>
  <header>
    <h1>
      Welcome 
      !
    </h1>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="logout-button"
        onClick={[Function]}
        type="submit"
      >
        Logout
      </button>
    </a>
  </header>
  <main
    className="welcome-page-main"
  >
    <div
      className="games-div"
    >
      <h2>
        Create a New Game
      </h2>
      <div>
        <form
          className="create-game-form"
          onSubmit={[Function]}
        >
          <label
            htmlFor="name"
          >
            Name
          </label>
          <input
            name="name"
            onChange={[Function]}
            required={true}
            value=""
          />
          <label
            htmlFor="notes"
          >
            Notes
          </label>
          <input
            name="notes"
            onChange={[Function]}
            value=""
          />
          <button
            className="submit"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <h2>
        's Saved Games
      </h2>
      <ul
        className="games-ul"
      />
    </div>
    <div
      className="players-div"
    >
      <div
        className="add-player-form"
      >
        <h2>
          Add New Player
        </h2>
        <form
          className="create-game-form"
          onSubmit={[Function]}
        >
          <label
            htmlFor="name"
          >
            Name
          </label>
          <input
            name="name"
            onChange={[Function]}
            required={true}
            value=""
          />
          <label
            htmlFor="notes"
          >
            Notes
          </label>
          <input
            name="notes"
            onChange={[Function]}
            value=""
          />
          <button
            className="submit"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <h2>
        's Players
      </h2>
      <div
        className="players-list-div"
      >
        <ul
          className="players-ul-welcome"
        />
      </div>
    </div>
  </main>
</div>
`;
